### PR TITLE
drivers: display: nhd_c12832a1z: initial commit

### PR DESCRIPTION
**Pull request by amiclaus**
Add support for NHD-C12832A1Z COG(Chip-On-Glass) Liquid Crystal Display Module.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: drivers/display/nhd_c12832a1z/nhd_c12832a1z.h [hunk 1]
```diff
@@ -0,0 +1,131 @@
+/***************************************************************************//**
+ *   @file   nhd_c12832a1z.h
+ *   @brief  Implementation of nhd_c12832a1z.h
```
Comment (danmois) [inline drivers/display/nhd_c12832a1z/nhd_c12832a1z.h:3]: Header file of

File: drivers/display/nhd_c12832a1z/nhd_c12832a1z.c [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+	uint8_t framebuffer_memory[4][128] = { 0 };
+	uint8_t page = 0xB0;
+	int32_t count = strlen(msg);
+	int32_t t_cursor = 0;
+
+	if ((t_cursor + count) > 128)
+		count = 128 - t_cursor;
+
+	for (j = 0; j < count; ++j) {
+		int cursor = (t_cursor + j) % 64;
```
Comment (danmois) [inline drivers/display/nhd_c12832a1z/nhd_c12832a1z.c:376]: Consider using macros for constants.

File: drivers/display/nhd_c12832a1z/nhd_c12832a1z.c [hunk 3]
```diff
@@ -0,0 +1,586 @@
...
+
+		for (i = 0; i < 8; i++)
+			framebuffer_memory[y][x+i] = ASC16[msg[cursor]][i];
+	}
+
+	ret = nhd_c12832a1z_write_cmd(dev, NHD_C12832A1Z_DISP_OFF);
+	if (ret)
+		return ret;
+
+	ret = nhd_c12832a1z_write_cmd(dev, 0x40); // Display start address + 0x40
```
Comment (danmois) [inline drivers/display/nhd_c12832a1z/nhd_c12832a1z.c:388]: Same as previous comment.

File: drivers/display/nhd_c12832a1z/nhd_c12832a1z.c [hunk 4]
```diff
@@ -0,0 +1,586 @@
...
+int nhd_c12832a1z_clear_lcd(struct nhd_c12832a1z_dev *dev)
+{
+	int ret;
+	unsigned int i;
+	uint8_t page = 0xB0;
+
+	ret = nhd_c12832a1z_write_cmd(dev, NHD_C12832A1Z_DISP_OFF);
+	if (ret)
+		return ret;
+
```
Comment (danmois) [inline drivers/display/nhd_c12832a1z/nhd_c12832a1z.c:435]: Extra line

File: drivers/display/nhd_c12832a1z/nhd_c12832a1z.c [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+ * @brief Initializes nhd_c12832a1z for display screening.
+ * @param device - The device structure.
+ * @param init_param - Initialization parameters.
+ * @return Returns 0 in case of success or negative error code otherwise.
+ */
+int nhd_c12832a1z_init(struct nhd_c12832a1z_dev **device,
+		       struct nhd_c12832a1z_init_param init_param)
+{
+	struct nhd_c12832a1z_dev *dev;
+	uint8_t command[3];
```
Comment (danmois) [inline drivers/display/nhd_c12832a1z/nhd_c12832a1z.c:475]: Unused variable.

**Review by amiclaus — COMMENTED** [review]
V2:
- add relevant macros where needed
- fix header comment
- remove unused variable.
- remove extra line

**Review by rbolboac — APPROVED** [review]

**Review by danmois — APPROVED** [review]

**Pull request merged**